diff --git a/lib/regcomp.c b/lib/regcomp.c
index 41b0f989c0..7e1f789b03 100644
--- a/lib/regcomp.c
+++ b/lib/regcomp.c
@@ -1762,7 +1762,7 @@ peek_token (re_token_t *token, re_string_t *input, reg_syntax_t syntax)
       token->mb_partial = 1;
       return 1;
     }
-  if (c == '\\')
+  if (!(syntax & RE_PLAIN) && c == '\\')
     {
       unsigned char c2;
       if (re_string_cur_idx (input) + 1 >= re_string_length (input))
@@ -1894,6 +1894,8 @@ peek_token (re_token_t *token, re_string_t *input, reg_syntax_t syntax)
   else
     token->word_char = IS_WORD_CHAR (token->opr.c);
 
+  if (!(syntax & RE_PLAIN))
+  {
   switch (c)
     {
     case '\n':
@@ -1965,6 +1967,7 @@ peek_token (re_token_t *token, re_string_t *input, reg_syntax_t syntax)
     default:
       break;
     }
+  }
   return 1;
 }
 
